### ISSUE TITLE
Build error with Edgex bridge

### ISSUE THREAD

**Issue by NeethuES-intel** [intel/video-analytics-serving#60]
Followed the instructions here to build edgex - https://github.com/intel/video-analytics-serving/tree/master/samples/edgex_bridge
Getting the following build error -
$ ./samples/edgex_bridge/fetch_edgex.sh
./samples/edgex_bridge/fetch_edgex.sh: line 6: cd: /home/nesubuntu207/Neethu/video-analytics-serving/samples/edgex_bridge/../../edgex: No such file or directory
Working folder: /home/nesubuntu207/Neethu/video-analytics-serving
3e2b35463a7ec6c6031721aeecb73f074c1f4926618851c9285d16ab174a00b0
dev_mqtt
Cloning into 'developer-scripts'...
remote: Enumerating objects: 36, done.
remote: Counting objects: 100% (36/36), done.
remote: Compressing objects: 100% (27/27), done.
remote: Total 2054 (delta 18), reused 9 (delta 7), pack-reused 2018
Receiving objects: 100% (2054/2054), 547.98 KiB | 1.40 MiB/s, done.
Resolving deltas: 100% (1417/1417), done.
./samples/edgex_bridge/fetch_edgex.sh: line 22: cd: ./developer-scripts/releases/nightly-build/compose-files/source: No such file or directory
make: *** No rule to make target 'compose'.  Stop.
cp: cannot stat '../docker-compose-nexus-no-secty-mqtt.yml': No such file or directory
Next steps:
./docker/build.sh
./docker/run.sh --dev
python3 samples/edgex_bridge/edgex_bridge.py --topic object_events --generate
docker-compose up -d
python3 samples/edgex_bridge/edgex_bridge.py --topic object_events

**Comment by NeethuES-intel** [comment]
I modified the links based on the fetch from latest edgex repo, but still missing some docker-compose files. Attaching the modified file here. Also attaching the build error with this version of the shell script
[fetch_edgex.txt](https://github.com/intel/video-analytics-serving/files/5998816/fetch_edgex.txt)

$ ./samples/edgex_bridge/fetch_edgex.sh
Working folder: /home/nesubuntu207/Neethu/video-analytics-serving_original/edgex
12e2479b2da60ae9ba0c35b7a7bfe12b668e9e31fd5466244e0c274fc9454ac0
dev_mqtt
fatal: destination path 'developer-scripts' already exists and is not an empty directory.
docker-compose -p edgex -f docker-compose-nexus-no-secty.yml up -d 
Starting edgex-core-consul ... done
Starting edgex-redis       ... done
Starting edgex-support-notifications ... done
Starting edgex-support-scheduler     ... done
Starting edgex-core-metadata         ... done
Starting edgex-core-data             ... done
Starting edgex-core-command          ... done
Starting edgex-device-rest                    ... done
Starting edgex-app-service-configurable-rules ... done
Starting edgex-sys-mgmt-agent                 ... done
Starting edgex-device-virtual                 ... done
Starting edgex-kuiper                         ... done
cp: cannot stat '../docker-compose-nexus-no-secty-mqtt.yml': No such file or directory
Next steps:
./docker/build.sh
./docker/run.sh --dev
python3 samples/edgex_bridge/edgex_bridge.py --topic object_events --generate
docker-compose up -d
python3 samples/edgex_bridge/edgex_bridge.py --topic object_events

**Comment by whbruce** [comment]
Thanks for logging issue and supplying updated fetch script. We have a fix and @tobiasmo1 will post steps for a workaround.

**Comment by tobiasmo1** [comment]
Hi @NeethuES-intel, 
Could you try again after you update `fetch_edgex.sh` (from line 19 forward) to replace with this block?

```
# Fetch EdgeX launch instructions
cd $VAS_SOURCE/edgex
EDGEX_DIR_REPO_DEVELOPER_SCRIPTS="./developer-scripts/"
if rm -Rf "$EDGEX_DIR_REPO_DEVELOPER_SCRIPTS"; then
    if git clone -b v1.3.1 https://github.com/edgexfoundry/developer-scripts.git $EDGEX_DIR_REPO_DEVELOPER_SCRIPTS; then
        cd "./$EDGEX_DIR_REPO_DEVELOPER_SCRIPTS/compose-builder/"
        if make compose no-secty ds-mqtt; then
            cp ../releases/hanoi/compose-files/docker-compose-hanoi-no-secty.yml $VAS_SOURCE/edgex/docker-compose.yml
            echo "Successfully fetched repo and produced compose file."
            cd $VAS_SOURCE
        else
            echo "ERROR making EdgeX compose file!"
            exit $?
        fi
    else
        echo "ERROR cloning $EDGEX_DIR_REPO_DEVELOPER_SCRIPTS repo!"
        exit $?
    fi
else
    echo "ERROR removing existing $EDGEX_DIR_REPO_DEVELOPER_SCRIPTS folder!"
    exit $?
fi
```

**Comment by NeethuES-intel** [comment]
Thanks @tobiasmo1. Build succeeded.
However when I tried the example for object detection, got the following error -
{"levelname": "ERROR", "asctime": "2021-02-22 22:48:53,045", "message": "Error on Pipeline 1: gst-resource-error-quark: Failed to connect to MQTT after maximum configured attempts. (3): /root/gst-video-analytics/gst/elements/gvametapublish/c_metapublish_mqtt.c(206): on_connect_failure (): /GstPipeline:pipeline7/GstGvaMetaPublish:destination", "module": "gstreamer_pipeline"}

Is it something to do with setting permission for MQTT ?
Attaching the complete message after running the following command - 
python3 ./samples/edgex_bridge/edgex_bridge.py --topic=objects_detected
[error_message.txt](https://github.com/intel/video-analytics-serving/files/6026657/error_message.txt)

With the curl command test, I did get 200 message back, however last command to check if event were received came []

nesubuntu207@nesubuntu207-desktop:~/Neethu/video-analytics-serving$ curl -i --get http://localhost:48081/api/v1/deviceservice/name/edgex-device-mqtt
HTTP/1.1 200 OK
Content-Type: application/json
Date: Tue, 23 Feb 2021 03:43:11 GMT
Content-Length: 556

{"created":1614051739841,"modified":1614051739841,"origin":1614051739840,"id":"9eda5803-417d-43da-9d5c-96ae597e9d54","name":"edgex-device-mqtt","operatingState":"ENABLED","addressable":{"created":1614051739839,"modified":1614051739839,"origin":1614051739835,"id":"758a006c-eafb-4266-8814-1f4876c957ac","name":"edgex-device-mqtt","protocol":"HTTP","method":"POST","address":"edgex-device-mqtt","port":49982,"path":"/api/v1/callback","baseURL":"http://edgex-device-mqtt:49982","url":"http://edgex-device-mqtt:49982/api/v1/callback"},"adminState":"UNLOCKED"}
nesubuntu207@nesubuntu207-desktop:~/Neethu/video-analytics-serving$ curl -i --get http://localhost:48081/api/v1/device/name/videoAnalytics-mqtt
HTTP/1.1 200 OK
Content-Type: application/json
Date: Tue, 23 Feb 2021 03:44:11 GMT
Content-Length: 1770

{"created":1614051739863,"modified":1614051739863,"origin":1614051739862,"description":"MQTT device that receives media analytics events.","id":"2e999705-a04a-4f8e-9456-c3270f25dbc2","name":"videoAnalytics-mqtt","adminState":"UNLOCKED","operatingState":"ENABLED","protocols":{"mqtt":{"ClientId":"videoAnalytics-pub","Host":"localhost","Password":"","Port":"1883","Schema":"tcp","Topic":"videoAnalyticsTopic","User":""}},"labels":["MQTT","VideoAnalyticsServing"],"service":{"created":1614051739841,"modified":1614051739841,"origin":1614051739840,"id":"9eda5803-417d-43da-9d5c-96ae597e9d54","name":"edgex-device-mqtt","operatingState":"ENABLED","addressable":{"created":1614051739839,"modified":1614051739839,"origin":1614051739835,"id":"758a006c-eafb-4266-8814-1f4876c957ac","name":"edgex-device-mqtt","protocol":"HTTP","method":"POST","address":"edgex-device-mqtt","port":49982,"path":"/api/v1/callback","baseURL":"http://edgex-device-mqtt:49982","url":"http://edgex-device-mqtt:49982/api/v1/callback"},"adminState":"UNLOCKED"},"profile":{"created":1614051739860,"modified":1614051739860,"description":"Device profile for inference events published by Video Analytics Serving over MQTT.","id":"aef55c61-1e88-4a42-b400-17acf68e445f","name":"videoAnalytics-mqtt","manufacturer":"VideoAnalyticsServing","model":"MQTT-2","labels":["MQTT","VideoAnalyticsServing"],"deviceResources":[{"description":"Inference with one or more detections on an analyzed media frame.","name":"videoAnalyticsData","properties":{"value":{"type":"String","readWrite":"R"},"units":{"type":"String","readWrite":"R"}}}],"deviceCommands":[{"name":"videoAnalyticsData","get":[{"operation":"get","object":"videoAnalyticsData","deviceResource":"videoAnalyticsData","parameter":"videoAnalyticsData"}]}]}}
nesubuntu207@nesubuntu207-desktop:~/Neethu/video-analytics-serving$ curl -i --get http://localhost:48080/api/v1/event/device/videoAnalytics-mqtt/100
HTTP/1.1 200 OK
Content-Type: application/json
Date: Tue, 23 Feb 2021 03:47:14 GMT
Content-Length: 3

[]

**Comment by tobiasmo1** [comment]
@NeethuES-intel - My apologies for the delay.  I notice dynamic addition of the mqtt broker was also improved in Hanoi release. So the fix is included in this replacement block below. It is nearly the same, but we add instruction to make compose with **mqtt-broker** and outputs to a new compose target.

With this you should see it functioning as expected. 
Let me know if this works for you.

If you encounter a problem, you may run ./clear_edgex.sh to reset persisted volumes.

```
# Fetch EdgeX launch instructions
cd $VAS_SOURCE/edgex
EDGEX_DIR_REPO_DEVELOPER_SCRIPTS="./developer-scripts/"
if rm -Rf "$EDGEX_DIR_REPO_DEVELOPER_SCRIPTS"; then
    if git clone -b v1.3.1 https://github.com/edgexfoundry/developer-scripts.git $EDGEX_DIR_REPO_DEVELOPER_SCRIPTS; then
        cd "./$EDGEX_DIR_REPO_DEVELOPER_SCRIPTS/compose-builder/"
        if make compose no-secty mqtt-broker ds-mqtt ; then
            cp ../releases/hanoi/compose-files/docker-compose-hanoi-no-secty-mqtt.yml $VAS_SOURCE/edgex/docker-compose.yml
            echo "Successfully fetched repo and produced compose file."
            cd $VAS_SOURCE
        else
            echo "ERROR making EdgeX compose file!"
            exit $?
        fi
    else
        echo "ERROR cloning $EDGEX_DIR_REPO_DEVELOPER_SCRIPTS repo!"
        exit $?
    fi
else
    echo "ERROR removing existing $EDGEX_DIR_REPO_DEVELOPER_SCRIPTS folder!"
    exit $?
fi

```

**Comment by NeethuES-intel** [comment]
@tobiasmo1 - Thanks
I am getting a different error now -
nesubuntu207@nesubuntu207-desktop:~/Neethu/video-analytics-serving$ python3 ./samples/edgex_bridge/edgex_bridge.py --topic=objects_detected
Traceback (most recent call last):
  File "./samples/edgex_bridge/edgex_bridge.py", line 12, in <module>
    from vaserving.vaserving import VAServing
ModuleNotFoundError: No module named 'vaserving'

However, I can import vaserving from terminal in a python prompt -
nesubuntu207@nesubuntu207-desktop:~/Neethu/video-analytics-serving$ python3
Python 3.8.5 (default, Jan 27 2021, 15:41:15) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from vaserving.vaserving import VAServing
>>>

**Comment by tobiasmo1** [comment]
@NeethuES-intel - This will occur if you attempt to run directly on your host. Be sure to run `edgex_bridge.py` from within our container to assure 'vaserving' module is accessible.  Did you skip this part of step #2 ?
```
./docker/run.sh --dev --name vaserving 
```

**Comment by NeethuES-intel** [comment]
Thank you @tobiasmo1. I missed that part. Yes, its working perfectly.

**Comment by whbruce** [comment]
Fixed in v0.5
